pandas/core/indexes/multi.py

```diff
@@ -1279,7 +1279,10 @@ def _get_grouper_for_level(self, mapper, level):
             # Remove unobserved levels from level_index
             level_index = level_index.take(uniques)
 
-        grouper = level_index.take(codes)
+        if not level_index.tolist():
```

Comment:
this is non performant and obscure
what are you trying to do?

Reply:
if index has garbage values like nan or none, make it possible to groupby. So check level_index,then empty just pass level_index

Reply:
@proost is the goal to check if level_index is non-empty?  i.e. could `if not len(level_index):` work?

Reply:
@jreback I think @proost is trying to resolve #20519, and needs some help to get a clue. (He chose this issue at PyConKR19 Sprint, and got lost in codes.)

Reply:
I'm reasonably confident that `if not len(level_index):` is the way to do this here

Reply:
@jbrockmendel  Thank you for your feedback. I change my code you mentioned.

Reply:
great.  can you rebase